.travis.yml

```diff
@@ -1,34 +1,58 @@
 language: php
-
-cache:
-    directories:
-        - $HOME/.composer/cache
-
 php:
     - 7.1
     - 7.2
-    - nightly
+    - 7.3
 
-matrix:
-    fast_finish: true
-    allow_failures:
-        - php: nightly
+before_install:
+  # turn off XDebug
+  - phpenv config-rm xdebug.ini || return 0
 
 install:
-    - export PATH="$PATH:$HOME/.composer/vendor/bin"
-    - composer global require php-coveralls/php-coveralls ^1.0.0
-    - composer update --no-interaction
-
-before_script:
-    - if php -v | grep -q 'Xdebug'; then coverage="--coverage-xml"; fi
-    - vendor/bin/codecept build
+  # Composer
+  - travis_retry composer install --no-progress --prefer-dist
 
 script:
-    - composer validate --strict
-    - composer outdated --direct --strict
-    - vendor/bin/php-cs-fixer fix --dry-run
-    - vendor/bin/phpstan analyse --level=max --configuration=phpstan.neon src tests
-    - vendor/bin/codecept run $coverage
-
-after_success:
-    - if [[ ! -z $coverage ]]; then coveralls; fi
+  # Nette/Tester
+  - composer run-script tester
+
+jobs:
+  include:
+    - env: title="Lowest Dependencies 7.1"
+      php: 7.2
+      install:
+        - travis_retry composer update --no-progress --prefer-dist --prefer-lowest
+      script:
+        - composer run-script tester
+
+    - stage: Quality Assurance
+      php: 7.2
+      script:
+        - composer run-script qa
+
+    - stage: Test Coverage
+      php: 7.2
+      script:
+        - composer run-script coverage
+      after_script:
+      - wget https://github.com/php-coveralls/php-coveralls/releases/download/v2.0.0/php-coveralls.phar
```

Comment:
This restricts the coverall library to the exact version 2.0.0. Wouldn't it be better to always get the latest? In Arachne I was doing it by `composer global require php-coveralls/php-coveralls ^1.0.0` (now it would be `^2.0.0`).

Reply:
You are right. I synchronized builds of all Contributte libraries recently, so I will make this change in all of them with next big update (nette 3.0). If you have any other ideas how to improve our builds then contact me please.